### PR TITLE
[SPARK-8576] Add spark-ec2 options to set IAM roles and instance-initiated shutdown behavior

### PR DESCRIPTION
**Pull request by nchammas**
Both of these options are useful when spark-ec2 is being used as part of an automated pipeline and the engineers want to minimize the need to pass around AWS keys for access to things like S3 (keys are replaced by the IAM role) and to be able to launch a cluster that can terminate itself cleanly.

### CODE REVIEW

**Comment by nchammas** [comment]
cc @JoshRosen

File: ec2/spark_ec2.py [hunk 1]
```diff
@@ -647,16 +655,19 @@ def launch_cluster(conn, opts, cluster_name):
...
+                    instance_type=opts.instance_type,
+                    placement=zone,
+                    min_count=num_slaves_this_zone,
+                    max_count=num_slaves_this_zone,
+                    block_device_map=block_map,
+                    subnet_id=opts.subnet_id,
+                    placement_group=opts.placement_group,
+                    user_data=user_data_content,
+                    instance_initiated_shutdown_behavior=opts.instance_initiated_shutdown_behavior,
+                    instance_profile_name=opts.instance_profile_name)
```
Comment (shivaram) [inline ec2/spark_ec2.py]: Just curious -- is boto happy if we pass in `None` here ?

Comment (nchammas) [inline ec2/spark_ec2.py]: `None` should be fine, but I'll double check that and empty string tomorrow.

Comment (nchammas) [inline ec2/spark_ec2.py]: Yup, both `None` (i.e. leaving out the option) and empty string work fine.

**Comment by shivaram** [comment]
Thanks for checking. LGTM

**Pull request closed without merging**